packages/plugin-ext/src/main/browser/tasks-main.ts

```diff
@@ -175,16 +175,30 @@ export class TasksMainImpl implements TasksMain, Disposable {
     }
 
     protected toTaskConfiguration(taskDto: TaskDto): TaskConfiguration {
-        return Object.assign(taskDto, {
-            _source: taskDto.source,
-            _scope: taskDto.scope
+        const { group, ...taskConfiguration } = taskDto;
+        if (group === 'build' || group === 'test') {
```

Comment:
A propos my comment regarding the possibility of accommodating the more elaborate `{kind: string}` form, if you use `TaskCustomization.isBuildTask` and `TaskCustomization.isTestTask` here as you do below, it will catch both the `string`s and any instances of the `object` type automatically.

Reply:
I am not sure if I understand the proposal, although the definition of group is actually aligned with vscode as follows, as a string, see: 
https://github.com/microsoft/vscode/blob/b37252c18238fffcebdc9fd7680f1dabea3a50a0/src/vs/workbench/api/common/shared/tasks.ts#L84-L97

Reply:
If we're confident that we'll never see anything but the string form, then you can ignore my suggestion; however, the `TaskCustomization.isBuildTask` and `TaskCustomization.isTestTask` checks are more general, and more likely to keep up with any future schema changes, so it's likely good to use them as widely as possible.

Reply:
The TaskCustomization.isBuildTask and TaskCustomization.isTestTask expect a TaskConfiguration as input, which is in the process to be converted to, that's why we are handling the conversion of group as a string separately so the Object.assign can apply.
Let me know if you see an alternative.

Reply:
That's true - though the code could work, the type check would be unhappy. Leave the code as is.